### DISCUSSION TITLE
Training not "working" on my own custom maps

### DISCUSSION THREAD

**Discussion by Rajivrocks-Ltd** [trackmania-rl/tmrl#98] [Q&A]
Hello yet again. I have made a very short custom map of my own to test the SAC example agent on. What I find tho that even after training for 3 hours on my GPU i still get a "train return" of 0. When I look at the graph on W&B it also reflects this. just a straight line at zero. the "episode_train_length" also is just stuck at 81 as far as I can telll. My first thought was, maybe this has something to do with the hyperparameters. I assume the hyperparameters in the config.json are specifically for the map that was provided with TMRL and that they won't really translate to good performance on other tracks.

Coming to the next point. I am testing a DuelingDQN agent as well. I tried to train it on my own map but the same thing happens. No training is happening "return train" is always 0 and the "episode_train_length" is also stuck at 81. But when I move back to the TMRL-test  map it somehow starts to train.

The only thing that I can think of which is honestly pretty logical, is that hyperparemeters need to be searchd for my own custom track.

Am I correct in my assumption that I need different hyper parameters for different tracks, and that the hyperparemters in the config.json are specifically tuned for this tmrl-test map that is provided?

**Comment by yannbouteiller** [comment]
Hi,

First things first, have you carefully checked that your reward function and observations make sense using `python -m tmrl --check-environment` ? What you describe sounds like a situation where the agent never gets any reward at all, which is most likely due to a faulty reward function.

You need to record a reward function for your track, it cannot work if you use the default reward function (which is for the `tmrl-train` map specifically). Check out the relevant section in the "Getting started" readme if you haven't done this yet.

**Reply by Rajivrocks-Ltd** [reply]
Oh okay, yeah this is the problem. I read the entire getting started readme but I probably looked over it. I'll have a look. Thanks for pointing this out!
